File: messaging-frontend/src/types/pagination.ts
export type Sort = {
    sorted: boolean;
    unsorted: boolean;
    empty: boolean;
}

export type Pageable = {
    sort: Sort;
    pageNumber: number;
    pageSize: number;
    offset: number;
    paged: boolean;
    unpaged: boolean;
}

export type PaginatedResponse<T> = {
    content: T[];
    pageable: Pageable;
    totalPages: number;
    totalElements: number;
    last: boolean;
    size: number;
    number: number;
    sort: Sort;
    numberOfElements: number;
    first: boolean;
    empty: boolean;
}

File: MessagingAndroid/app/src/main/java/ma/tayeb/messaging_android/types/Pagination.kt
package ma.tayeb.messaging_android.types

data class Sort(
    val unsorted: Boolean,
    val sorted: Boolean,
    val empty: Boolean
)

data class Pageable(
    val sort: Sort,
    val pageNumber: Int,
    val pageSize: Int,
    val offset: Int,
    val paged: Boolean,
    val unpaged: Boolean
)

data class PaginatedResponse<T>(
    val content: List<T>,
    val pageable: Pageable,
    val totalPages: Int,
    val totalElements: Long,
    val last: Boolean,
    val size: Int,
    val number: Int,
    val sort: Sort,
    val numberOfElements: Int,
    val first: Boolean,
    val empty: Boolean
)
